cluster/src/dunit/scala/io/snappydata/cluster/SnappySecureJob.scala

```diff
@@ -0,0 +1,243 @@
+    tableData.foreach(pw.println)
+
+    pw.println("****Done****")
+  }
+
+  def accessAndModifyTablesOwnedByOthers(sns: SnappySession, config: Config): Unit = {
+    pw.println()
+    pw.println("****Accessing other user's tables****")
+
+    val otherColTab = config.getString(otherColTabName)
+    val otherRowTab = config.getString(otherRowTabName)
+    val grantedOp = config.getString(opCode)
+
+    val opCodeToSQLs = Map("select" -> Seq(s"SELECT * FROM $otherColTab",
+        s"SELECT * FROM $otherRowTab"),
+      "insert" -> Seq(s"INSERT INTO $otherColTab VALUES (1, 'ONE', 1.1)",
+        s"INSERT INTO $otherRowTab VALUES (1, 'ONE', 1.1)"),
+      "update" -> Seq(s"UPDATE $otherColTab SET COL1 = 100 WHERE COL1 = 1",
+        s"UPDATE $otherRowTab SET COL1 = 100 WHERE COL1 = 1"),
+      "delete" -> Seq(s"DELETE FROM $otherColTab WHERE COL1 = 100",
+        s"DELETE FROM $otherRowTab WHERE COL1 = 100"))
+
+    grantedOp match {
+      case "nogrant" => opCodeToSQLs.keys.foreach(k => opCodeToSQLs(k).foreach(s =>
+        assertGrantRevoke(s, k, granted = false)))
+      case op: String => opCodeToSQLs(op).foreach(s => assertGrantRevoke(s, op, granted = true))
+    }
+
+    def assertGrantRevoke(s: String, op: String, granted: Boolean = false): Unit = {
+      if (granted) {
+        sns.sql(s).collect()
+        pw.println(s"Success for $s")
+      } else {
+        try {
+          sns.sql(s).collect()
+          assert(false, s"Should have failed $s")
+        } catch {
+          case t: Throwable if (t.getMessage.contains(s"does not have ${op.toUpperCase} " +
+              s"permission on") || t.getMessage.contains(s"does not have SELECT permission on")) =>
+            pw.println(s"Found expected exception for $s")
+            // t.getStackTrace.foreach(s => pw.println(s"${t.getMessage}\n  ${s.toString}"))
+          case t: Throwable => pw.println(s"UNEXPECTED ERROR FOR $s:\n[${t.getMessage}]")
+            t.getStackTrace.foreach(s => pw.println(s"  ${s.toString}"))
+            throw t
+        }
+      }
+    }
+
+    pw.println("****Done****")
+  }
+
+}
+
+object SnappyStreamingSecureJob extends SnappyStreamingJob {
+
+  def verifySessionAndConfig(snSession: SnappySession, jobConfig: Config): Unit = {
+    assert(snSession.conf.getOption(Attribute.USERNAME_ATTR).isDefined, "Username not set in conf")
+    assert(snSession.conf.getOption(Attribute.PASSWORD_ATTR).isDefined, "Password not set in conf")
+    try {
+      jobConfig.getString(Constant.STORE_PROPERTY_PREFIX + com.pivotal.gemfirexd
+          .Attribute.USERNAME_ATTR)
+      assert(false, "Boot credentials set in job config")
+    } catch {
+      case _: ConfigException.Missing => // expected
+    }
+    try {
+      jobConfig.getString(Constant.STORE_PROPERTY_PREFIX + com.pivotal.gemfirexd
+          .Attribute.PASSWORD_ATTR)
+      assert(false, "Boot credentials set in job config")
+    } catch {
+      case _: ConfigException.Missing => // expected
+    }
+  }
+
```

Comment:
Why separate job types for this? I don't think it is a good idea to ask users to implement separate job types. If the job configuration contains credentials then set into session else not. Jobs will automatically fail on a secure cluster when running operations that need credentials.

Reply:
Which job types are you referring to?
The utility method verifySessionAndConfig() (I could put it somewhere else) simply verifies that 1) the session made available to the job has the credentials set into it and 2) job config does not have boot credentials inadvertently passed to it.
This utility method is called from each of the test snappy jobs (i.e. SnappySecureJob, SnappyStreamingJob, SnappyJavaSecureJob and SnappyJavaStreamingSecureJob).

Reply:
Confused by this. Why are these new job types under tests? Do we expect users to use security this way by implementing new job types?

Reply:
These new jobs **are not** meant users but unit tests only. In fact, how users implement snappy jobs will not change with security.
I have added these jobs only for the unit testing purpose. I could have used the jobs already present in the examples folder but I needed to do some specific checks hence wrote these new ones.
The credentials supplied by the user while submitting his/her job will be set into the snappysession instance made available to the job. This was already taken care of, for SnappySQLJob but was missing for SnappyStreamingJob and JavaSnappyStreamingJob.

Reply:
@ashetkar if this is the only way in which credentials with jobs are being tested, then is it enough? There should be some tests that mirror how users will do it.